creator-node/src/services/stateMachineManager/stateReconciliation/SecondarySyncHealthTracker.ts

```diff
@@ -1,332 +1,301 @@
 /**
  * SecondarySyncHealthTracker
- * API for Primary to measure SyncRequest success and failure counts per Secondary, User, and Day
+ * API for Primary to track wallet-secondary encountered errors. Used to determine if further
+ * action should be taking place on a secondary for a wallet
  */
 
-import type { WalletsToSecondariesMapping } from '../types'
+import { ComputeWalletOnSecondaryUserInfo } from '../stateMonitoring/types'
+import {
+  SYNC_ERRORS_TO_MAX_NUMBER_OF_RETRIES,
+  SYNC_ERRORS_TO_MAX_NUMBER_OF_RETRIES_MAP
+} from './SecondarySyncHealthTrackerConstants'
+import {
+  WalletToSecondaryAndMaxErrorReached,
+  WalletToSecondaryToExceedsMaxErrorsAllowed
+} from './types'
+
+const { logger: genericLogger } = require('../../../logging')
+const redisClient = require('../../../redis')
 
-// eslint-disable-next-line import/no-unresolved
-import { UserSecondarySyncMetricsMap } from '../stateMonitoring/types'
+const REDIS_KEY_PREFIX_PRIMARY_TO_SECONDARY_SYNC_FAILURE =
+  'PRIMARY_TO_SECONDARY_SYNC_FAILURE'
 
-const config = require('../../../config')
-const redisClient = require('../../../redis')
-const { logger } = require('../../../logging')
+const REDIS_KEY_EXPIRY_SECONDS = 259_200 // 3 days -- small range of buffer time to delete keys by
+const PROCESS_USERS_BATCH_SIZE = 30
 
-const RedisKeyPrefix = 'SecondarySyncRequestOutcomes-Daily'
+export class SecondarySyncHealthTracker {
+  walletToSecondaryToExceedsMaxErrorsAllowed: WalletToSecondaryToExceedsMaxErrorsAllowed
+  walletsToSecondaryAndMaxErrorReached: WalletToSecondaryAndMaxErrorReached
 
-const DailyRedisKeyExpirationSec =
-  90 /* days */ * 24 /* hr */ * 60 /* min */ * 60 /* s */
+  constructor(
```

Comment:
it took me a minute to realize that the `walletToSecondaryToExceedsMaxErrorsAllowed` passed in here is the same result that's returned from `computeWalletOnSecondaryExceedsMaxErrorsAllowed()`. it would be helpful to have a comment here noting that passing in a value to the constructor is a shortcut for initializing with an empty constructor and then calling `computeWalletOnSecondaryExceedsMaxErrorsAllowed()` with an array of wallets